File: frontend/src/types/Book.ts
// This file defines the TypeScript interface for a Book object, which includes properties such as bookID, title, author, publisher, isbn, classification, category, pageCount, and price.
export interface Book {
    bookID: number;
    title: string;
    author: string;
    publisher: string;
    isbn: string;
    classification: string;
    category: string;
    pageCount: number;
    price: number;
}

File: backend/WebApplication1/WebApplication1/Data/Book.cs
namespace WebApplication1.Data
{
    public class Book
    {
        public int BookID { get; set; }
        public required string Title { get; set; }
        public required string Author { get; set; }
        public required string Publisher { get; set; }
        public required string ISBN { get; set; }
        public required string Classification { get; set; }
        public required string Category { get; set; }
        public int PageCount { get; set; }
        public decimal Price { get; set; }
    }
}
